bokeh/plotting/helpers.py

```diff
@@ -453,10 +479,32 @@ def _process_active_tools(toolbar, tool_map, active_drag, active_scroll, active_
     GlyphRenderer
 """
 
+def _add_sigfunc_info(func, argspecs, glyphclass, extra_docs):
+    func.__name__ = glyphclass.__name__.lower()
+
+    kwlines = []
+    kws = glyphclass.properties() - set(argspecs)
+    for kw in sorted(kws):
+        prop = getattr(glyphclass, kw)
+        if prop.__doc__:
+            typ = prop.__class__.__name__
+            desc = " ".join(x.strip() for x in prop.__doc__.strip().split("\n\n")[0].split('\n'))
+        else:
+            typ = str(prop)
+            desc = ""
+        kwlines.append(_arg_template % (kw, typ, desc, prop.class_default(glyphclass)))
+
+    arglines = []
+    for arg, spec in argspecs.items():
+        arglines.append(_arg_template % (arg, spec['type'], spec['desc'], spec['default']))
+
+    func.__doc__ = _doc_template % (func.__name__, "\n".join(arglines), "\n".join(kwlines))
+    if extra_docs:
+        func.__doc__ += extra_docs
```

Comment:
This adds `__name__` and `__doc__` to the wrapper. Anything else? Can't use `functools.wraps` because that seems to demolish the signature at least in python 3.x (I think there was a feature added to make the reported signature overridable, and `wraps` seems to use it, resulting in `*args, **kw` signature all over again)